src/Symfony/Component/Lock/Store/RedisStore.php

```diff
@@ -46,6 +54,88 @@ public function __construct($redisClient, $initialTtl = 300.0)
     }
 
     /**
+     * Creates a Redis connection using a DSN configuration.
+     *
+     * Example DSN:
+     *   - redis://localhost
+     *   - redis://example.com:1234
+     *   - redis://secret@example.com/13
+     *   - redis:///var/run/redis.sock
+     *   - redis://secret@/var/run/redis.sock/13
+     *
+     * @param string $dsn
+     * @param array  $options See self::$defaultConnectionOptions
+     *
+     * @throws InvalidArgumentException When the DSN is invalid.
+     *
+     * @return \Redis|\Predis\Client According to the "class" option
+     */
+    public static function createConnection($dsn, array $options = array())
```

Comment:
We could maybe consider a new component, named `Dsn`, with factories for any kind of DSN we want to support?
not sure we can make it for 3.4, but what do others think about the idea?